src/dfcx_scrapi/tools/google_drive_connector.py

```diff
@@ -0,0 +1,196 @@
+"""Utility file for Google Sheets"""
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import time
+
+import google.auth
+import gspread
+from gspread_dataframe import set_with_dataframe
+from oauth2client.service_account import ServiceAccountCredentials
+import pandas as pd
+
+
+class GoogleDriveConnector:
+    """Class for google sheets operations"""
+
+    global_scopes = [
+        "https://spreadsheets.google.com/feeds",
+        "https://www.googleapis.com/auth/drive",
+    ]
+
+    def __init__(
+        self, creds_path: str = None, creds_dict: dict = None, scope=False
+    ):
+        scopes = GoogleDriveConnector.global_scopes
+        if scope:
+            scopes += scope
+
+        if creds_path:
+            creds = ServiceAccountCredentials.from_json_keyfile_name(
+                filename=creds_path, scopes=scopes
+            )
+        elif creds_dict:
+            creds = ServiceAccountCredentials.from_json_keyfile_dict(
+                keyfile_dict=creds_dict, scopes=scopes
+            )
+        else:
+            creds = google.auth.default(scopes=scopes)[0]
+
+        self.client = gspread.authorize(creds)
+
+    def create_google_sheet(self, sheet_name: str):
+        """Create new google sheet object, this sheet is created
+        under service account drive account and must be shared to an account
+        to view it from that account.
+
+        Args:
+          sheet_name: name of the sheets object to create.
+        """
+        self.client.create(title=sheet_name)
+
+    def delete_google_sheet(self, sheet_name: str):
+        """Delete a google sheet object
+
+        Args:
+          sheet_name: name of the sheets object to delete.
+        """
+        g_sheets = self.client.open(sheet_name)
+        self.client.del_spreadsheet(g_sheets.id)
```

Comment:
@Hgithubacct Does the `self.client.del_spreadsheet()` provide a log message back to the user stating the Sheet was successfully deleted?

If not can we add some kind of confirmation?

Reply:
@vudoy Please confirm this action when you run through unit testing.